addons/uom/static/src/components/many2x_uom_tags/many2x_uom_tags.js

```diff
@@ -13,6 +13,15 @@ import { UomAutoComplete } from "@uom/components/uom_autocomplete/uom_autocomple
 import { roundPrecision } from "@web/core/utils/numbers";
 import { onWillUpdateProps } from "@odoo/owl";
 
+function _getProductRelatedModel() {
+    const field = this.env.model.config.fields[this.props.productField]
+    let resModel = field?.relation;
```

Comment:
This approach doesn't handle the case when the opened form is `product.template` or `product.product` form itself. In that case, the value of `productField` will be the fallback value: `product_id` (because the option `product_field` is not set on the widget in `product.product` & `product.template` forms). So here, it will try to access `product_id` on `product.template` or `product.product` which doesn't exist, which will lead to an error in the next check, although it shouldn't. We could do something like:
```suggestion
    let resModel = field?.relation ?? this.env.model.config.resModel;
```

Reply:
@mohammadhossam I saw that, sorry I already did a modification in the meantime.
The previous version wasn't neither, it just assumed it always was a `product.product` (since it was the fallback value) no matter what, so the widget didn't cause a traceback but was silently wrong.
That said, for what I saw, with no quantity field this widget lost half its interest, is it really useful to use it in the product's form view ?

Reply:
Yes, it's still important to show the widget even if there's no related quantity field because we still want to show the factors relative to the product base unit.